lisp/ess-inf.el

```diff
@@ -172,97 +172,56 @@ This may be useful for debugging."
-           (cur-dir (inferior-ess--maybe-prompt-startup-directory procname temp-dialect))
-           (default-directory cur-dir)
-           buf method)
-      (ess-write-to-dribble-buffer
-       (format "(inf-ess 1.1): procname=%s temp-dialect=%s, buf-name=%s \n"
-               procname temp-dialect buf-name-str))
-
-      (cond
-       ;; 1) try to use current buffer, if inferior-ess-mode but no process
-       ((and (not (comint-check-proc (current-buffer)))
-             (derived-mode-p 'inferior-ess-mode))
-        (setq buf (current-buffer))
-        ;; don't change existing buffer name in this case; It is very
-        ;; commong to restart the process in the same buffer.
-        (setq buf-name-str (buffer-name))
-        (setq method 1))
-
-       ;; 2)  Take the *R:N* buffer if already exists (and contains dead proc!)
-       ;; fixme: buffer name might have been changed, iterate over all
-       ;; inferior-ess buffers
-       ((get-buffer buf-name-str)
-        (setq buf (get-buffer buf-name-str))
-        (setq method 2))
-
-       ;; 3)  Pick up a transcript file or create a new buffer
-       (t
-        (setq buf (if ess-ask-about-transfile
-                      (let ((transfilename (read-file-name "Use transcript file (default none):"
-                                                           cur-dir
-                                                           "")))
-                        (if (string= transfilename "")
-                            (get-buffer-create buf-name-str)
-                          (find-file-noselect (expand-file-name  transfilename))))
-                    (get-buffer-create buf-name-str)))
-        (setq method 3)))
-
-      (ess-write-to-dribble-buffer
-       (format "(inf-ess 2.0) Method #%d start=%s buf=%s\n" method cur-dir buf))
-
-      (set-buffer buf)
+           (proc-name (let ((ntry 1))
+                        (while (get-process (ess-proc-name ntry temp-dialect))
+                          (setq ntry (1+ ntry)))
+                        (ess-proc-name ntry temp-dialect)))
+           (proc (get-process proc-name))
+           (inf-buf (inferior-ess--get-proc-buffer-create proc-name))
+           (inf-name (buffer-name inf-buf))
+           (cur-dir (inferior-ess--maybe-prompt-startup-directory proc-name temp-dialect))
+           (default-directory cur-dir))
+
+      (set-buffer inf-buf)
       (set 'default-directory cur-dir)
       ;; TODO: Get rid of this, we should rely on modes to set the
       ;; variables they need.
       (ess-setq-vars-local ess-customize-alist)
 
-      (ess-write-to-dribble-buffer
-       (format "(inf-ess 2.1): ess-language=%s, ess-dialect=%s buf=%s \n"
-               ess-language  ess-dialect (current-buffer)))
-
-      (let* ((infargs (or ess-start-args
-                          inferior-ess-start-args))
-             (proc (get-process procname)))
+      (let ((inf-args (or ess-start-args
+                          inferior-ess-start-args)))
```

Comment:
This variable (inferior-ess-start-args) should probably just be removed. It's a little awkward, and disconnected from e.g. inferior-R-args, which is awkward.

Reply:
I think it's never reached, at least for R as ess-start-args is always nil. I agree, it's bad design. Let's remove.

Reply:
Best done in another PR as this seems to affect all languages.